pennylane/ops/qubit/pcphase_decomposition.md

```diff
@@ -0,0 +1,216 @@
+
+# Algorithm description
+
+For this decomposition, we will decompose the generator of the `PCPhase` gate,
+which we denote as $G$, into multiple projectors, which generate (multi-controlled)
+phase shift gates. The generator of `PCPhase` is given as 
+
+
+$$
+G = 2 \left(\sum_{j=0}^{d-1} |j\rangle\langle j|\right) - \mathbb{I}_N
+= \text{diag}(
+\underset{d\text{ times}}{\underbrace{1, \dots, 1}},
+\underset{(N-d)\text{ times}}{\underbrace{-1, \dots, -1}}
+),
+$$
+
+where we denoted the overall dimension as $N=2^n$ for $n$ qubits and the
+subspace dimension as $d$, which is called `dim` in code.
+Our first step is to move to a slightly different convention in order to arrive at a
+*projector* that has ones and zeros, instead of ones and
+minus ones (note how $G$ is not a projector because $G^2=\mathbb{I}_N\neq G$).
+To do this, we define
+
+$$
+\Pi = \begin{cases}
+    \sum_{j=0}^{d-1} |j\rangle\langle j| & \text{for }\ d\leq \tfrac{N}{2}\\
+    \sum_{j=d}^{N-1} |j\rangle\langle j| & \text{for }\ d> \tfrac{N}{2}
+\end{cases},
+$$
+
+so that $G=2\Pi -\mathbb{I}_N$ for $d\leq \tfrac{N}{2}$ and
+$G=\mathbb{I}_N - 2 \Pi$ for $d>\tfrac{N}{2}$. We may summarize
+this as $G=\sigma (2\Pi -\mathbb{I}_N)$ with the sign $\sigma=\pm 1$
+defined accordingly for the two cases.
+
+Now we want to decompose $\Pi$, a sum of the computational basis state projectors
+$|j\rangle\langle j|$ onto the $d$ consecutive smallest (largest, for
+$\sigma=-1$) basis states,
+into projectors that contain a power of two basis state projectors. We will see below that
+these are exactly the generators of (multi-controlled) phase shift gates, allowing us
+to implement the decomposition. As we may flip the sign of these individual phase shift
+gates, we may also use subtraction when decomposing $\Pi$.
+
+This decomposition of $\Pi$ is based on decomposing the integer $d$ into sums
+and differences of powers of two, which is detailed in a separate section below. This
+decomposition reads
+
+$$
+d = \sum_{i=0}^{n-1} c_i 2^{n-1-i}, \quad c_i\in\{-1, 0, 1\}.
+$$
+
+Now we need to use this decomposition to combine projectors of $2^{n-1-i}$
+computational basis state projectors into the desired $\Pi$.
+We start with the scenario $\sigma=+1$, i.e., $d\leq \tfrac{N}{2}$, in which
+case this decomposition has the form
+
+$$
+\Pi = \sum_{i=0}^{n-1} c_i \sum_{j=s_i}^{e_i} |j\rangle\langle j|,
+$$
+
+where $s_i$ and $e_i$ are the start and end positions
+for the $i$\ th contribution to $\Pi$.
+Besides computing the integer decomposition itself, this decomposition function is mostly
+about figuring out the right control structure to realize the correct positioning
+according to $s_i$ and $e_i$.
+
+To begin, initialize empty lists of control wires and control values, corresponding
+to $s_0=0$. Then, starting at $i=0$, which also marks the current wire,
+go through the integer decomposition $\{c_i\}_i$ from above and perform the
```

Comment:
Probably unintentional I think, otherwise, I don't understand the notation. 

```suggestion
go through the integer decomposition $\{c_i\}$ from above and perform the
```

Reply:
Ah, yeah, this is sometimes done to indicate that the set runs over the index $i$.
Pedantically, the object $\{c_i\}$ contains exactly one element, potentially with an underspecified index $i$, and $\{c_i\}_{i=low}^{high}$, or $\{c_i\}_i$ for short, is a set of $high-low$ elements...
I'm happy to skip it, it's pedantic!